src/redux/wallets.js

```diff
@@ -224,6 +227,54 @@ export const createAccountForWallet = (id, color, name) => async (
   return newWallets;
 };
 
+export const getWalletENSAvatars = async (walletsState, dispatch) => {
+  const { wallets, walletNames, selected } = walletsState;
+  const walletKeys = Object.keys(wallets);
+  let updatedWallets;
+  for (const key of walletKeys) {
+    const wallet = wallets[key];
+    const addresses = [];
+    let avatarChanged = false;
+    for (const account of wallet?.addresses) {
+      const ens =
+        (await fetchReverseRecord(account.address)) ||
+        walletNames[account.address];
+
+      if (ens) {
+        const images = await fetchImages(ens);
+        avatarChanged = true;
+        addresses.push({
+          ...account,
+          image:
+            typeof images?.avatarUrl === 'string' &&
+            images?.avatarUrl !== account?.image
+              ? images?.avatarUrl
+              : account.image,
+        });
+      } else {
+        addresses.push(account);
+      }
+    }
+    // don't update wallets if nothing changed
+    if (avatarChanged) {
```

Comment:
https://github.com/rainbow-me/rainbow/pull/3374/files#r877656269

now we go though the addresses once and set the avatarUrl to the correct account